build/source/driver/multi_driver.f90

```diff
@@ -830,170 +822,46 @@ program multi_driver
  end if  ! if the first time step
 
  ! ****************************************************************************
- ! *** loop through HRUs and GRUs
+ ! *** model simulation
  ! ****************************************************************************
 
- ! initialize variables
+ ! loopp through GRUs
  do iGRU=1,nGRU
 
-  ! initialize runoff variables
-  bvarStruct%gru(iGRU)%var(iLookBVAR%basin__SurfaceRunoff)%dat(1)    = 0._dp  ! surface runoff (m s-1)
-  bvarStruct%gru(iGRU)%var(iLookBVAR%basin__ColumnOutflow)%dat(1)    = 0._dp  ! outflow from all "outlet" HRUs (those with no downstream HRU)
+  ! simulation for a single GRU
+  call run_oneGRU(&
+                  ! model control
+                  iGRU,               & ! intent(in):    GRU index
```

Comment:
Can you define this so it cannot be changed inside the function? (see comment below in the `run_oneGRU` procedure)